lib/dpl/provider/elastic_beanstalk.rb

```diff
@@ -39,7 +39,11 @@ def push_app
         create_bucket unless bucket_exists?
 
         if options[:zip_file]
-          zip_file = File.join(Dir.pwd, options[:zip_file])
+          if File.dirname(options[:zip_file]) == "."
```

Comment:
@BanzaiMan @Galindo-janrain thanks for the feedback.

From reading the documentation I learned that `File.absolute_path` and `File.expand_path` both produce the absolute_path and behave almost exactly the same. The only difference is that `File.expand_path` will also expand `~` to be the value of `HOME` Environmental Variable.

I will use `File.expand_path` because handling `~` seams useful.

### File.absolute_path(file_name [, dir_string] ) → abs_file_name

https://ruby-doc.org/core-1.9.3/File.html#method-c-absolute_path

```
Converts a pathname to an absolute pathname. Relative paths are referenced from the current working directory of the process unless dir_string is given, in which case it will be used as the starting point. If the given pathname starts with a “~” it is NOT expanded, it is treated as a normal directory name.
```

### File.expand_path(file_name [, dir_string] ) → abs_file_name click to toggle source

https://ruby-doc.org/core-1.9.3/File.html#method-c-expand_path

```
Converts a pathname to an absolute pathname. Relative paths are referenced from the current working directory of the process unless dir_string is given, in which case it will be used as the starting point. The given pathname may start with a “~”, which expands to the process owner’s home directory (the environment variable HOME must be set correctly). “~user” expands to the named user’s home directory.
```